### DISCUSSION TITLE
Problem in accessing client-supported modules

### DISCUSSION THREAD

**Discussion by 15IITian** [fedimint/fedimint#4204] [Q&A] [answered]
To access client-supported modules -> I try to take `modules` field from the client variable of type `Client` -> as `modules: ClientModuleRegistry` which has a function `iter-modules` to iterate all  modules.

![image](https://github.com/fedimint/fedimint/assets/133089376/7847be11-7e1e-4bd2-a5e1-788e343d65b7)

But `Client` struct fields are private and I found no such function in its implementation to iterate over modules .
Is there any other way to access them ?

**Comment by elsirion** [comment]
I don't think you need access to that necessarily. You could call `get_config` and iterate over module instances and then get the module client via `get_module_client_dyn` to check if it's supported.

Maybe there's a better way to do it if you had access to that field? Why don't you have access anyway? Are you implementing the logic in the cli binary? I'd probably just define a fn on the client that does the heavy lifting and then just call that from `fedimint-cli`.

**Reply by 15IITian** [reply]
> Why don't you have access anyway? Are you implementing the logic in the cli binary?

It's because the Client struct is defined inside fedimint-client/lib.rs file but I am accessing it inside fedimint-cli/lib.rs  so I am able to access the struct as a whole (as whole struct is made public not its individual fields)

**Comment by 15IITian** [comment]
> You could call get_config and iterate over module instances and then get the module client via get_module_client_dyn to check if it's supported.

That gives the modules which are federation supported and may be for client , right?
but is it possible that there exists some modules which are client supported but not federation supported inside `ClientModuleRegistry` .
If yes, don't we  have to list them ?

**Reply by elsirion** [reply, marked as answer]
DPC made a good point in the original issue that listing client-supporte but not federation-supported modules makes no sense. So this isn't an issue. Also please don't open discussions for questions like this. Use discord or maybe open an issue. If you don't understand the client enough maybe look at some simpler issues that you can tackle on your own without lots of help from others.

**Reply by 15IITian** [reply]
Ok
